### PR TITLE
Chore: Use `catalogs` to manage versions of common dependencies

### LINKED ISSUE #2807: NEXT: Update Svelte/React packages to utilize PNPM Catalogs for version syncing
Following our recent [integration of Zag.js](https://github.com/skeletonlabs/skeleton/pull/2778), we've learned that each individual Zag component is installed as it's own separate package per framework. For example:

```
"@zag-js/accordion": "^0.65.0",
"@zag-js/avatar": "^0.65.0",
"@zag-js/progress": "^0.65.0",
"@zag-js/radio-group": "^0.65.0",
"@zag-js/rating-group": "^0.65.0",
"@zag-js/react": "^0.65.0",
"@zag-js/switch": "^0.65.0",
"@zag-js/tabs": "^0.65.0",
```

These versions must stay in sync to ensure these components remain operational and type data resolves as expected.

@Hugos68 and @AdrianGonz97 have determined a potential solution for this using [PNPM catalogs](https://pnpm.io/catalogs). This will provide a variable in place of the version number that we can control within the monorepo and ensure these remain in sync.

If necessary we should update any related contributor documentation around this. Specifying a minimum version of PNPM to use, etc.

## Goals:

- Implement PNPM catalogs for Zag in our Svelte/React packages
- Ensure all Skeleton team members make local updates (update PNPM, etc)
- Document requirements for contributors

### PR DESCRIPTION
**Pull request by Hugos68**
## Linked Issue

Closes #2807 

## Description

Uses a monorepo root `catalog` to define versions for every package so it's easier to upgrade, manage, etc.
See: https://pnpm.io/catalogs

> Note for contributors: this will bump PNPM to require a minimum version of 9.5.0 or higher!

## Changsets

Instructions: Changesets automate our changelog. If you modify files in `/packages`, run `pnpm changeset` in the root of the monorepo, follow the prompts, then commit the markdown file. Changes that add features should be `minor` while chores and bugfixes should be `patch`. Please prefix the changeset message with `feat:`, `bugfix:` or `chore:`.

## Checklist

Please read and apply all [contribution requirements](https://www.skeleton.dev/docs/contributing).

- [x] This PR targets the `dev` branch (NEVER `master`)
- [x] Documentation reflects all relevant changes
- [x] Branch is prefixed with: `docs/`, `feat/`, `chore/`, `bugfix/`
- [x] Ensure Svelte and Typescript linting is current - run `pnpm ci:check`
- [x] Ensure Prettier linting is current - run `pnpm format`
- [x] All test cases are passing - run `pnpm test`
- [x] Includes a changeset (if relevant; see above)

### CODE REVIEW

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+  '@zag-js/react': ^0.65.0
+
+  # Machines
+  '@zag-js/accordion': ^0.65.0
+  '@zag-js/avatar': ^0.65.0
+  '@zag-js/rating-group': ^0.65.0
+  '@zag-js/tabs': ^0.65.0
+  '@zag-js/progress': ^0.65.0
+  '@zag-js/radio-group': ^0.65.0
+  '@zag-js/switch': ^0.65.0
```
Comment (AdrianGonz97) [inline pnpm-workspace.yaml:17]: we may also want to add a `typescript` to the catalog too given that it's used in every package

**Review by AdrianGonz97 — COMMENTED** [review]
we'll also want to update the `engines` and `packageManager` field in the root `package.json` to use `9.6.0` or higher (the latest is `9.7.1`).

File: pnpm-workspace.yaml (hunk 1 above)
Comment (Hugos68) [reply pnpm-workspace.yaml:17]: I was wondering this too, Chris agreed in the discord channel so I'll go ahead and add it too. Should we do latest stable? `5.5.4`

Comment (AdrianGonz97) [reply pnpm-workspace.yaml:17]: yea, updating to the latest should be fine

Comment (Hugos68) [reply pnpm-workspace.yaml]: Done.

Comment (endigo9740) [reply pnpm-workspace.yaml]: @Hugos68 when you feel this is complete and ready for review, go ahead and mark "ready for review" and add Adrian as a reviewer please.

**Comment by AdrianGonz97** [comment]
this looks good now, but we'll first have to resolve why vercel's deployment is failing before we merge

**Pull request merged**